tests/repository_simulator.py

```diff
@@ -136,10 +176,30 @@ def fetch(self, url: str) -> Iterator[bytes]:
                 version = None
                 role = parts[0]
             yield self._fetch_metadata(role, version)
+        elif spliturl.path.startswith("/targets/"):
+            # figure out the actual target path and the hash prefix
+            path = spliturl.path[len("/targets/") :]
+            dir_parts, sep , prefixed_filename = path.rpartition("/")
+            prefix, _, filename = prefixed_filename.partition(".")
+            target_path = f"{dir_parts}{sep}{filename}"
+
+            yield self._fetch_target(target_path, prefix)
         else:
             raise FetcherHTTPError(f"Unknown path '{spliturl.path}'", 404)
 
+    def _fetch_target(self, target_path: str, hash: Optional[str]) -> bytes:
+        """Return data for 'target_path', checking 'hash' if it is given"""
```

Comment:
Can you please mention that hash will be requested if consistent snapshot -> consistent targets' naming is used? Took me some time to understand why the hash check is done here. Although when you see metadata and version you can get the idea ... so maybe it's my problem only :)

Reply:
added some docstrings and a NotImplementedError for `not consistent_snapshot` to make this clearer